File: webapp/frontend/src/codes.js
// This file contains the code for the different algorithms in
// Leap and the different types of functions that are available.

// Algo Code
export const COUNT_ALGO = "count_fn"
export const PRIVATE_SITE_COUNT_ALGO = "count_fn_site_dp" // Adds privacy at site layer
export const PRIVATE_CLOUD_COUNT_ALGO = "count_fn_cloud_dp" // Adds privacy at cloud layer
export const SUM_ALGO = "sum_fn"
export const VARIANCE_ALGO = "var_fn"
export const FEDERATED_LEARNING_ALGO = "fl_fn"
export const QUANTILE_ALGO = "quantile_fn"

// Leap Types
export const UDF = 1
export const LAPLACE_UDF = 2
export const EXPONENTIAL_UDF = 3
export const PREDEFINED = 4
export const PRIVATE_PREDEFINED = 5
export const FEDERATED_LEARNING = 6

File: api/codes.py
# This file contains the code for the different algorithms in
# Leap and the different types of functions that are available.

# Algo Code
COUNT_ALGO = "count_fn"
COUNT_ALGO_RC = "count_fn_redcap"
COUNT_ALGO_RC_QUERY = "count_fn_redcap_query"
PRIVATE_SITE_COUNT_ALGO = "count_fn_site_dp" # Adds privacy at site layer
PRIVATE_CLOUD_COUNT_ALGO = "count_fn_cloud_dp" # Adds privacy at cloud layer
SUM_ALGO = "sum_fn"
VARIANCE_ALGO = "var_fn"
FEDERATED_LEARNING_ALGO = "fl_fn"
QUANTILE_ALGO = "quantile_fn"
LOG_REG = "log_reg_fn"

# Leap Types
UDF = 1
LAPLACE_UDF = 2
EXPONENTIAL_UDF = 3
PREDEFINED = 4
PRIVATE_PREDEFINED = 5
FEDERATED_LEARNING = 6